cli-tests/tests/pmm-cli/server/install.spec.ts

```diff
@@ -1,61 +1,63 @@
-    const volumeName = 'pmm-data-install-test'
-
-    try {
-      let output = await cli.exec(`
-        pmm server docker install 
-          --json
-          --admin-password=${adminPw}
-          --docker-image="${imageName}"
-          --https-listen-port=1443
-          --http-listen-port=1080
-          --container-name=${containerName}
-          --volume-name=${volumeName}`
-      )
-
-      // Output
-      await output.assertSuccess()
-      expect(output.stderr).toContain('Starting PMM Server')
-
-      // http client
-      let client = new PMMRestClient('admin', adminPw, 1080)
-      await client.works()
-
-      // https client
-      client = new PMMRestClient('admin', adminPw, 1443, {
-        baseURL: 'https://localhost:1443',
-        ignoreHTTPSErrors: true,
-      })
-      let resp = await client.doPost('/v1/Settings/Get')
-      let respBody = await resp.json()
-
-      expect(resp.ok()).toBeTruthy()
-      expect(respBody).toHaveProperty('settings')
-
-      // Container name      
-      output = await cli.exec(`docker ps --format="{{.Names}}" | grep "^${containerName}$" | wc -l`)
-      expect(output.stdout.trim()).toEqual('1')
-
-      // Volume name
-      output = await cli.exec(`docker volume ls --format="{{.Name}}" | grep "^${volumeName}$" | wc -l`)
-      expect(output.stdout.trim()).toEqual('1')
-
-      // Docker image
-      output = await cli.exec(`docker ps --format="{{.Names}} {{.Image}}" | grep -E "^${containerName} ${imageName}$" | wc -l`)
-      expect(output.stdout.trim()).toEqual('1')
-    } finally {
-      await teardown([`^${containerName}$`], [volumeName])
-    }
-  })
-})
+import { test, expect } from '@playwright/test';
+import * as cli from '@helpers/cliHelper';
+import PMMRestClient from '@tests/support/types/request';
+import { teardown } from '@tests/helpers/containers';
+
+test.describe.configure({ mode: 'parallel' });
+test.describe('PMM Server Install tests', async () => {
+  const adminPassword = 'admin123';
+  const containerName = 'pmm-server-install-test';
+  const imageName = 'percona/pmm-server:2.32.0';
+  const volumeName = 'pmm-data-install-test';
+
+  test.afterAll(async ({}) => {
+    await teardown([`^${containerName}$`], [volumeName]);
+  });
+
+  test('"pmm server docker install" respects relevant flags', async ({ }) => {
+    // TODO: add "docker pull" test images as tests configuration to speed up tests
+    let output = await cli.exec(`
+      pmm server docker install 
+        --json
+        --admin-password=${adminPassword}
+        --docker-image="${imageName}"
+        --https-listen-port=1443
+        --http-listen-port=1080
+        --container-name=${containerName}
+        --volume-name=${volumeName}`);
+    await output.assertSuccess();
+
+    await expect(async () => {
+      expect(output.stderr).toContain('Starting PMM Server');
+    }).toPass({
+      // Probe, wait 1s, probe, wait 2s, probe, wait 2s, probe, wait 2s, probe, ....
```

Comment:
Why do we need to retry here? I believe the command has already finished, correct?

Reply:
It has failed on "parallel" command run on this check. So it has been left as is - it will exit on the 1st try and will apply reties only on fail. Though I left is as a safe block to move for any test structure.

Reply:
The code appears to be trying to find `Starting PMM Server` string in the `output.stderr` which does not change when it's retried.
Am I missing anything?

Reply:
No, you are not missing anything. Please see the original code added by you: https://github.com/percona/pmm/blob/main/cli-tests/tests/pmm-cli/server/install.spec.ts#L29
Also please see the logs: https://github.com/percona/pmm/actions/runs/4830926936/jobs/8607722662
Here is a linked screenshot: https://www.screencast.com/t/bjYMvObHdrd
Green section is "stdout" pipe and red section is "stderr" pipe. pmm binary makes non-error output in the error pipe.
Would you like me to rise a defect for this?

Reply:
It should not be stderr, not sure why it is. Let's open a ticket for it.
The original comment is still valid though - I don't think retrying this part adds any value since the output is always the same during retries.

Reply:
PMM-12084 - added ticket. And removed retries.